universe/Universe.cpp

```diff
@@ -1660,6 +1660,21 @@ void Universe::ForgetKnownObject(int empire_id, int object_id) {
     objects.Remove(object_id);
 }
 
+void Universe::CheckShipDesignVisibility(int empire_id, int object_id, Visibility vis) {
```

Comment:
That would be even more misleading.

Maybe MakeDesignVisibileToEmpireIfShipVisible or somesuch?